faker/providers/person/pl_PL/__init__.py

```diff
@@ -774,3 +774,26 @@ def pwz_nurse(self, kind='nurse'):
         kind_char = 'A' if kind == 'midwife' else 'P'
 
         return '{:02d}{}{}'.format(region, ''.join(map(str, core)), kind_char)
+
+    def nip(self):
+        """
+        Returns 10 digit of Number of tax identification.
+        Polish: Numer identyfikacji podatkowej (NIP).
+
+        https://pl.wikipedia.org/wiki/NIP
+
+        """
+        nip = []
+        for _ in range(9):
+            nip.append(self.random_digit())
+
+        weights = (6, 5, 7, 2, 3, 4, 5, 6, 7)
+        check_sum = sum(d * w for d, w in zip(nip, weights)) % 11
+
+        if check_sum % 11 == 10:
+            nip.append(0)
```

Comment:
I used [google translate](https://translate.google.com/#view=home&op=translate&sl=auto&tl=en&text=Dziesi%C4%85ta%20cyfra%20NIP%20jest%20cyfr%C4%85%20kontroln%C4%85%2C%20obliczan%C4%85%20wed%C5%82ug%20poni%C5%BCszego%20algorytmu%3A%0A%0APomno%C5%BCy%C4%87%20ka%C5%BCd%C4%85%20z%20pierwszych%20dziewi%C4%99ciu%20cyfr%20odpowiednio%20przez%20wagi%3A%206%2C%205%2C%207%2C%202%2C%203%2C%204%2C%205%2C%206%2C%207%2C%0AZsumowa%C4%87%20wyniki%20mno%C5%BCenia%2C%0AObliczy%C4%87%20reszt%C4%99%20z%20dzielenia%20przez%2011%20(operacja%20modulo%2011).%0ANIP%20jest%20tak%20generowany%2C%20aby%20nigdy%20w%20wyniku%20tego%20dzielenia%2C%20jako%20reszta%2C%20nie%20uzyska%C4%87%20liczby%2010.%20Zgodnie%20z%20tym%20algorytmem%20numer%20000-000-00-00%20jest%20prawid%C5%82owy%2C%20ale%20nie%20ma%20sensu.%20Dla%20ci%C4%85gu%20cyfr%20123-456-78-90%20nie%20mo%C5%BCna%20dobra%C4%87%20cyfry%20kontrolnej%2C%20by%20wygenerowa%C4%87%20prawid%C5%82owy%20NIP.), and it said that the check sum value cannot be 10. If that is truly the case, then the current implementation will generate invalid numbers.

Instead of calling this function recursively till the check sum does not equal to 10 (~9.1% chance of happening per attempt), I suggest using this algorithm instead when the check sum ends up being 10:

1. Increment/decrement a random digit by 1 (whichever is possible) 
2. Add/subtract the corresponding weight for that digit to the current check sum (which will be 10)
3. Apply modulo 11 to the new check sum, and the result will be the new check digit

For example, the number `123456789` will have a check sum of 10. If you want to alter the first digit from the left, say we increment it by 1, our number will become 223456789. The first digit's weight is 6, and since we incremented the first digit, we add it such that new check digit will be `(10 + 6) % 11 = 5`, so the new NIP will be `2234567895`.

Reply:
Do you think it would be beneficial do double check result NIP with code, for validation 
https://pl.wikibooks.org/wiki/Kody_%C5%BAr%C3%B3d%C5%82owe/Implementacja_NIP
?

Reply:
I had this nagging feeling that I was forgetting something, and thank you for reminding me. You will need to write some tests. This is where the validation will be performed to ensure that `nip()` is generating valid output.

In your tests, you can probably generate 100 samples and validate each of them. I chose 100, because if any 9 digit combination has ~9.1% chance of returning a check sum of 10, then there is less than a 0.01% chance that all 100 samples will pass validation should the algorithm be flawed.

Reply:
I have added tests, and notice that actual there still is bug with generation, I will have to look deeper for this issue

Reply:
I think now everything should be OK, corner case handled. Maybe so code optimization is required, waiting for comments